src/napari/_vispy/utils/text.py

```diff
@@ -71,3 +73,168 @@ def _has_visible_text(layer: Points | Shapes) -> bool:
     ):
         return False
     return len(layer._indices_view) != 0
+
+
+def get_text_width_height(text: Text) -> tuple[float, float]:
+    """Get the width and height of a vispy text visual in screen pixels."""
+    if isinstance(text.text, str):
+        strings = [text.text]
+    elif isinstance(text.text, list):
+        strings = text.text
+    else:
+        raise TypeError('Text should either be a string or a list of strings')
+
+    top_left_corners = []
+    bottom_right_corners = []
+    for string in strings:
+        if string == '':
+            continue
+
+        # this is a private vispy function that calculates the Vertex Buffer Object
+        # for the text to be sent to the shaders. This normallly only happens on
+        # the fly when rendering, but we need it here because it allows us to
+        # calculate the bounding box of the text as it would be when rendered
+        buffer = _text_to_vbo(
+            string, text._font, *text._anchors, text._font._lowres_size
+        )
+
+        pos = buffer['a_position']
+        top_left_corners.append(pos.min(axis=0))
+        bottom_right_corners.append(pos.max(axis=0))
+
+    top_left = np.min(top_left_corners, axis=0) if top_left_corners else (0, 0)
+    bottom_right = (
+        np.max(bottom_right_corners, axis=0)
+        if bottom_right_corners
+        else (0, 0)
+    )
+
+    font_size = get_text_font_size(text)
+
+    # these magic numbers (1.2 and 1.3) are from trial and error
+    return (bottom_right[0] - top_left[0]) * font_size * 1.3, (
+        bottom_right[1] - top_left[1]
+    ) * font_size * 1.2
+
+
+def get_text_font_size(text: Text) -> float:
+    """Get the logical font size of a text visual, rescaled by dpi."""
+    return text.font_size * getattr(text.canvas, 'pixel_size', 1)
+
```

Comment:
> If this were the case (unlike now) it might clear up a lot of my issues with overlay text.

@TimMonko then the issues you have are actually causes by us trying to do something in the first place. Vispy already handles scaling based on the QT settings.

If you remove the rescaling part of this line, does it behave as you expect? @psobolewskiPhD what about you?

```suggestion
def get_text_font_size(text: Text) -> float:
    """Get the logical font size of a text visual, rescaled by dpi."""
    return text.font_size
```

Reply:
I'm feeling really confused right now. I've been waiting to review this until I was at my desk with a monitor easy to change the scaling on. I've also tested on both Qt5 and Qt6 (since they behave differently wrt scaling on windows), but observe no difference.
https://github.com/napari/napari/pull/7836#issuecomment-3304020645 was done from SciPy on my laptop with 1.0 scaling. 

Now, on my desktop (surprisingly also 1.0 scaling...) I currently get:
<img width="490" height="550" alt="image" src="https://github.com/user-attachments/assets/c27ef31d-26ea-4e05-b19b-918c87ce7057" />

And removing the suggested part I see no change. But thats seemingly because text.canvas.pixel_size doesn't exist, so its always returning 1. I can test this by changing the default return and see that it only ever uses 1...
<img width="490" height="551" alt="image" src="https://github.com/user-attachments/assets/a1f56334-475f-4766-975d-895955d91279" />

I guess that is functionally doing nothing? Not sure why the text is misplaced then on my laptop -- I'll test tonight to see if the pixel_size attr exists on there, because if I DO change it to 0.8 I get extremely similar results on my desktop

<img width="950" height="554" alt="image" src="https://github.com/user-attachments/assets/ab573d31-99fe-4213-95ed-a1887b6704b2" />

If I change scaling to 2.0 I get the same appearance... and there is still no text.canvas.pixel_size
<img width="1662" height="1060" alt="image" src="https://github.com/user-attachments/assets/4497bba7-c7bf-4165-aa39-edd7b98e1c53" />

Reply:
So you could access `text.canvas.native.devicePixelRatio()` .. but the behavior looks no different even if that is 2 (thus changing the calculated size) ... so might need to take a step back and make sure things are in order. Sorry I don't have more time now.

<img width="958" height="1060" alt="image" src="https://github.com/user-attachments/assets/427fc5df-9f50-4316-8e1f-725c28b76655" />

Reply:
At the very least, the current behavior I'm seeing on this monitor (where text gets bigger as the UI gets bigger) is what I want to see.

Reply:
Oof, my bad, it was supposed to be `pixel_scale`, not `pixel_size` :sweat: somehow I switched without noticing...

But either way, if this behaviouur is what you wanted, then we don't actually need to to anything special like we were trying to do, we can remove all this hidpi code :P

Reply:
It should work as intended now.